projects/rero/ng-core/src/lib/record/detail/detail.component.ts

```diff
@@ -166,10 +176,26 @@ export class DetailComponent implements OnInit, OnDestroy {
   /**
    * Delete the record and go back to previous page.
    * @param event - DOM event
-   * @param pid - string, PID to remove
-   */
-  deleteRecord(pid: string) {
-    this._recordUiService.deleteRecord(this._route.snapshot.paramMap.get('type'), pid).subscribe((result: any) => {
+   * @param element - string (PID to remove) or object
+   */
+  deleteRecord(element: any) {
+    const pid = typeof element === 'object' ? element.metadata.pid : element;
+    if (typeof element === 'object') {
+      if (this._config.redirectUrl) {
```

Comment:
You can merge both `if` statements. Should be more readable (my point of view)

Reply:
done